### PR TITLE
[Core] Missing template on MathUtils

### PR DESCRIPTION
**Pull request by loumalouomega**

### CODE REVIEW

**Comment by loumalouomega** [comment]
Ping

**Comment by philbucher** [comment]
I am not the exepert here 
@roigcarlo can you please have a look?

File: kratos/utilities/math_utils.h [hunk 1]
```diff
@@ -1215,43 +1215,46 @@ class MathUtils
      * \f$ [ s11, s22, s12 ] \f$ for 2D case.
      * @param rStressVector the given stress vector
      * @return the corresponding stress tensor in matrix form
+     * @tparam TVector The vector type considered
+     * @tparam TMatrixType The matrix returning type
      */
-    static inline MatrixType StressVectorToTensor(const Vector& rStressVector)
+   template<class TVector, class TMatrixType = MatrixType>
```
Comment (roigcarlo) [inline kratos/utilities/math_utils.h]: One more space please.

Comment (roigcarlo) [reply kratos/utilities/math_utils.h]: Also, to maintain compatibility, shouldn't be:

```C++
template<class TVector = Vector, class TMatrix = MatrixType>
```

And also, to keep the style, if we call a template `TVector` the other type should be `TMatrix`, or `TVectorType` and `TMatrixType`

**Review by roigcarlo — CHANGES_REQUESTED** [review]
Some minor changes (and a question).

Also please don't change the name of the variables in the middle of another change as is difficult enough to read...

File: kratos/utilities/math_utils.h [hunk 2]
```diff
@@ -1215,43 +1215,46 @@ class MathUtils
...
+        TMatrixType stress_tensor;
 
         if (rStressVector.size()==3) {
-            StressTensor.resize(2,2,false);
-            StressTensor(0,0) = rStressVector[0];
-            StressTensor(0,1) = rStressVector[2];
-            StressTensor(1,0) = rStressVector[2];
-            StressTensor(1,1) = rStressVector[1];
+            stress_tensor.resize(2,2,false);
+            stress_tensor(0,0) = rStressVector(0);
```
Comment (roigcarlo) [inline kratos/utilities/math_utils.h:1229]: Why the change from `[]` to `()`? Shouldn't be the same or has something to do with the AMatrix PR?

File: kratos/utilities/math_utils.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/utilities/math_utils.h]: No, because the first type is explictely detected

Comment (loumalouomega) [reply kratos/utilities/math_utils.h]: Implicitily

File: kratos/utilities/math_utils.h (hunk 2 above)
Comment (loumalouomega) [reply kratos/utilities/math_utils.h:1229]: This is the way to be compatible with AMatrix and to work always with all the arrays/vectors. Trust me I tried

**Comment by loumalouomega** [comment]
imagen

File: kratos/utilities/math_utils.h (hunk 1 above)
Comment (roigcarlo) [reply kratos/utilities/math_utils.h]: But we are adding the possibility to use this function with any type having [] operator. 

For example I can define 

```C++ 
class foo {
    int& operator[] (int x) {
        return 0;
    }
}
``` 

As a vector, and it will be perfectly fine to call it with this new class. Not saying its wrong, but I am not sure we intend to open the function this way.

**Review by roigcarlo — APPROVED** [review]
Any way code is fine so :+1:
